onebusaway-android/src/main/java/org/onebusaway/android/nav/NavigationServiceProvider.java

```diff
@@ -706,4 +716,113 @@ private void silence(long duration, int queueFlag) {
             mTTS.playSilence(duration, queueFlag, null);
         }
     }
+
+    public void getUserFeedback() {
+        //TODO - Log "Yes" or "No" including plaintext feedback using Firebase Analytics
+
+        Application app = Application.get();
+        NotificationCompat.Builder mBuilder;
+
+        String message = Application.get().getString(R.string.feedback_notify_dialog_msg);
+        mFirstFeedback = Application.getPrefs().getBoolean(FIRST_FEEDBACK, true);
+
+        // Create delete intent to set flag for snackbar creation next time the app is opened.
+        Intent rIntent = new Intent(app.getApplicationContext(), FeedbackReceiver.class);
+        rIntent.putExtra(FeedbackReceiver.NOTIFICATION_ID, NOTIFICATION_ID + 1);
+
+        if ((mFirstFeedback) || (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) || (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P)) {
+
+            Intent fdIntent = new Intent(app.getApplicationContext(), FeedbackActivity.class);
+            fdIntent.setAction(REPLY_ACTION);
+            fdIntent.putExtra("CallingAction", "no");
+            fdIntent.putExtra("NotificationId", NOTIFICATION_ID + 1);
+            fdIntent.putExtra("TripId", mTripId);
+            PendingIntent fdPendingIntentNo = PendingIntent.getActivity(app.getApplicationContext()
+                    , 1, fdIntent,PendingIntent.FLAG_UPDATE_CURRENT);
+
+            fdIntent = new Intent(app.getApplicationContext(), FeedbackActivity.class);
+            fdIntent.setAction(REPLY_ACTION);
+            fdIntent.putExtra("CallingAction", "yes");
+            PendingIntent fdPendingIntentYes = PendingIntent.getActivity(app.getApplicationContext()
+                    , 2, fdIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+
+            rIntent.putExtra(FeedbackReceiver.ACTION_NUM,
```

Comment:
Instead of setting this as an extra it should be the Intent action:
`intent.setAction(...)`

Reply:
Per further discussion, let's leave these as extras and just rename them from actions to "response", or something similar.

Reply:
This is the delete intent, so it should have the action of `ACTION_DISMISS_FEEDBACK`.

Reply:
We have created a delete action and modified the code to set it as an action of the intent.